File: sdk/src/main/java/com/cxense/cxensesdk/model/UserProfileGroup.java
package com.cxense.cxensesdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Group object for item in {@link UserProfile}
 *
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */
@SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
public final class UserProfileGroup {
    @JsonProperty("group")
    String group;
    @JsonProperty("count")
    int count;
    @JsonProperty("weight")
    double weight;

    /**
     * Gets the category or type of information.
     *
     * @return category
     */
    public String getGroup() {
        return group;
    }

    /**
     * Gets count
     *
     * @return count
     */
    public int getCount() {
        return count;
    }

    /**
     * Gets the relative prominence of the item/group combination.
     *
     * @return weight value
     */
    public double getWeight() {
        return weight;
    }
}

File: sdk/src/main/java/io/piano/android/cxense/model/UserProfileGroup.kt
package io.piano.android.cxense.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Group object for item in {@link UserProfile}
 *
 */
@JsonClass(generateAdapter = true)
public class UserProfileGroup(
    @Json(name = "group") public val group: String?,
    @Json(name = "count") public val count: Int,
    @Json(name = "weight") public val weight: Double,
)
